agent/structs/config_entry_gateways.go

```diff
@@ -231,11 +239,49 @@ func (e *IngressGatewayConfigEntry) validateServiceSDS(lis IngressListener, svc
 	return nil
 }
 
+func validateGatewayTLSConfig(tlsCfg GatewayTLSConfig) error {
+	if tlsCfg.TLSMinVersion != types.TLSVersionUnspecified {
+		if err := types.ValidateTLSVersion(tlsCfg.TLSMinVersion); err != nil {
+			return err
+		}
+	}
+
+	if tlsCfg.TLSMaxVersion != types.TLSVersionUnspecified {
+		if err := types.ValidateTLSVersion(tlsCfg.TLSMaxVersion); err != nil {
+			return err
+		}
+	}
+
+	// TODO: Is there any reasonable way to validate that TLSMaxVersion is
+	// greater than or equal to TLSMinVersion as string types?
+
+	if len(tlsCfg.CipherSuites) != 0 {
+		// TODO: is there any contextual way to determine Envoy's default TLS min and max versions if either
+		// TLSMinVersion or TLSMaxVersion are undefined or set to TLS_AUTO?
+		if tlsCfg.TLSMinVersion == types.TLSv1_3 {
+			return fmt.Errorf("configuring CipherSuites is only applicable to conncetions negotiated with TLS 1.2 or earlier, TLSMinVersion is set to %s", tlsCfg.TLSMinVersion)
+		}
+
+		// TODO: is it possible to emit a warning but not return an error from here if TLSMaxVersion is undefined,
```

Comment:
No, we don't currently have a mechanism to return warnings up from config entry write operations unfortunately.